src/renault_api/cli/charge/schedule.py

```diff
@@ -0,0 +1,189 @@
+"""CLI function for a vehicle."""
+import re
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import aiohttp
+import click
+from tabulate import tabulate
+
+from renault_api.cli import helpers
+from renault_api.cli import renault_vehicle
+from renault_api.kamereon.helpers import DAYS_OF_WEEK
+from renault_api.kamereon.models import ChargeDaySchedule
+from renault_api.kamereon.models import ChargeSchedule
+
+
+_DAY_SCHEDULE_REGEX = re.compile(
+    "(?P<prefix>T?)"
+    "(?P<hours>[0-2][0-9])"
+    ":"
+    "(?P<minutes>[0-5][0-9])"
+    "(?P<suffix>Z?)"
+    ","
+    "(?P<duration>[0-9]+)"
+)
+
+
+@click.group()
+def schedule() -> None:
+    """Display or update charging schedules."""
+    pass
+
+
+@schedule.command()
+@click.option("--id", type=int, help="Schedule ID")
+@click.pass_obj
+@helpers.coro_with_websession
+async def show(
+    ctx_data: Dict[str, Any],
+    *,
+    id: Optional[int] = None,
+    websession: aiohttp.ClientSession,
+    **kwargs: Any,
+) -> None:
+    """Display charging settings."""
+    vehicle = await renault_vehicle.get_vehicle(
+        websession=websession, ctx_data=ctx_data
+    )
+    response = await vehicle.get_charging_settings()
+
+    # Display mode
+    click.echo(f"Mode: {response.mode}")
+    if not response.schedules:  # pragma: no cover
+        click.echo("No schedules found.")
+        return
+
+    for schedule in response.schedules:
+        if id and id != schedule.id:
+            continue
+        click.echo(
+            f"\nSchedule ID: {schedule.id}{' [Active]' if schedule.activated else ''}"
+        )
+
+        headers = [
+            "Day",
+            "Start time",
+            "End time",
+            "Duration",
+        ]
+        click.echo(
+            tabulate(
+                [_format_charge_schedule(schedule, key) for key in DAYS_OF_WEEK],
+                headers=headers,
+            )
+        )
+
+
+def _format_charge_schedule(schedule: ChargeSchedule, key: str) -> List[str]:
+    details: Optional[ChargeDaySchedule] = getattr(schedule, key)
+    if not details:  # pragma: no cover
+        return [key, "-", "-", "-"]
+    return [
+        key.capitalize(),
+        helpers.get_display_value(details.startTime, "tztime"),
+        helpers.get_display_value(details.get_end_time(), "tztime"),
+        helpers.get_display_value(details.duration, "minutes"),
+    ]
+
+
+@schedule.command()
+@click.option("--id", type=int, help="Schedule ID")
```

Comment:
How about making `id` a (mandatory) argument? it’s required every time anyways, and `renault-api charge schedule set 3 --monday 12:15,120` looks kinda neat IMO

Reply:
I hesitated about it, but it can't be compulsory on "show".
So do we remove it completely from "show" (ie. always show all schedules)?
Or do we keep it as an option of "show" and an argument on "set"?

Reply:
Ah … you are right about that. I would personally tend to drop it on `show`, as my gut feeling is that in most situations, most users of the CLI are not inconvenienced seeing all schedules (they are well formatted after all …). But even when keeping it as option on `show`, I think it would be still nice to have it as argument on `set`. Just my personal gut feeling though …